src/brogue/IO.c

```diff
@@ -1256,7 +1256,10 @@ void getCellAppearance(short x, short y, enum displayGlyph *returnChar, color *r
                    && (playerCanSeeOrSense(x, y) || ((monst->info.flags & MONST_IMMOBILE) && (pmap[x][y].flags & DISCOVERED)))
                    && (!monsterIsHidden(monst, &player) || rogue.playbackOmniscience)) {
             needDistinctness = true;
-            if (player.status[STATUS_HALLUCINATING] > 0 && !(monst->info.flags & (MONST_INANIMATE | MONST_INVULNERABLE)) && !rogue.playbackOmniscience) {
+            if (player.status[STATUS_HALLUCINATING] > 0
+                    && !(monst->info.flags & (MONST_INANIMATE | MONST_INVULNERABLE))
+                    && !rogue.playbackOmniscience
+                    && !(pmap[x][y].flags & TELEPATHIC_VISIBLE)) {
```

Comment:
any reason not to check for player telepathic status here, instead of telepathic visibility?

Reply:
ok, i dunno what the issue was before, but switching it here worked fine. on seed 3 which has toads and a telepathy potion, i tested hallu running out while telepathy was active, and telepathy running out while hallu was active, and everything seems fine.